rfc/text/advanced/ap_35_authentication_cryptosign.md

```diff
@@ -1,3 +1,1124 @@
+2. **CCR-2**: The `realm` must match for all certificates to the respective realm.
+3. **CCR-3**: The type of the first certificate in the chain must be a `EIP712DelegateCertificate`, and all subsequent certificates must be of type `EIP712AuthorityCertificate`.
+4. **CCR-4**: The last certificate must be self-signed (`issuer` equals `subject`), it is a root CA certificate.
+5. **CCR-5**: The intermediate certificate's `issuer` must be equal to the `subject` of the previous certificate.
+6. **CCR-6**: The root certificate must be `validFrom` before the intermediate certificate
+7. **CCR-7**: The `capabilities` of intermediate certificate must be a subset of the root cert
+8. **CCR-8**: The intermediate certificate's `subject` must be the delegate certificate `delegate`
+9. **CCR-9**: The intermediate certificate must be `validFrom` before the delegate certificate
+10. **CCR-10**: The root certificate's signature must be valid and signed by the root certificate's `issuer`.
+11. **CCR-11**: The intermediate certificate's signature must be valid and signed by the intermediate certificate's `issuer`.
+12. **CCR-12**: The delegate certificate's signature must be valid and signed by the `delegate`.
+
+
+#### Trustroots
+
+Certificate chains allow to verify a delegate certificate following the Issuers-Subjects up to a *Root CA*, which is a self-signed certificate (issuer and subject are identical). The *Root CA* represents the *Trustroot* of all involved delegates.
+
+When both a connecting WAMP client and the WAMP router are using the same *Root CA* and thus use a common
+*Trustroot*, they are said to be authorized in the same trust domain (identified by the trustroot).
+There are two types of *Root CAs* and *Trustroots*:
+
+1. *Standalone Trustroot*
+2. *On-chain Trustroot*
+
+A *Standalone Trustroot* is managed by a single operator/owner, does not allow infrastructure elements
+(nodes, client, realms) to be integrated between different operators/owners and is privately stored on the
+respective operators systems only, usually as files or in databases.
+
+An *On-chain Trustroot* in contrast is stored in Ethereum and publically shared between different
+operators/owners which allows infrastructure elements (nodes, clients, realms) to be integrated.
+For example, clients/nodes operated by different operators can authenticate to each other and
+nodes operated by different operators can authenticate to each other sharing the hosting of one realm.
+
+The management of *On-Chain Trustroots* depends on the policy of the trustroot which is chosen
+and fixed when the trustroot is created:
+
+1. *Open*
+2. *Permitted*
+3. *Private*
+
+With an *Open On-chain Trustroot*, new certificates can be added to a certificate chain freely
+and only requires a signature by the respective intermediate CA issuer.
+
+##### Standalone Trustroots
+
+For a *Standalone Trustroot* the `trustroot` MUST be specified in `HELLO.Details.authextra.trustroot|string`
```

Comment:
I think, there is no description of what is the value for `trustroot` field?  Is it some hex encoded string of root certificate fingerprint or something else?
For on-chain trustroot — there is a clear description, that it is eth address on public blockchain where root certificate is stored.